spec/models/miq_server/log_management_spec.rb

```diff
@@ -1,10 +1,91 @@
 require "spec_helper"
 
+def stub_vmdb_util_methods_for_collection_log
+  VMDB::Util.stub(:zip_logs)
+  VMDB::Util.stub(:compressed_log_patterns).and_return(["log/evm.1122.gz"])
+  VMDB::Util.stub(:get_evm_log_for_date).and_return("20151209_141429_20151217_140845")
+  VMDB::Util.stub(:get_log_start_end_times).and_return([Time.zone.now, Time.zone.now])
+end
+
+shared_examples "post_[type_of_log]_logs" do |type, type_of_log|
+  it "uses #{type} file_depot in LogFile for upload" do
+    @zone.log_file_depot = zone_depot
+    @miq_server.log_file_depot = server_depot
+
+    stub_vmdb_util_methods_for_collection_log
+    allow_any_instance_of(LogFile).to receive(:upload)
+
+    method = "post_#{type_of_log}_logs".to_sym
+    @miq_server.send(method, miq_task.id, @miq_server.log_depot(type))
+    log_file_depot = LogFile.first.file_depot
+
+    if type == "Zone"
+      expect(log_file_depot).to eq(zone_depot)
+      expect(log_file_depot).not_to eq(server_depot)
+    else
+      expect(log_file_depot).not_to eq(zone_depot)
+      expect(log_file_depot).to eq(server_depot)
+    end
+
+    expect(LogFile.first.miq_task).to eq(miq_task)
+  end
+end
+
+shared_examples "post_logs_uses_depot" do |is_zone_depot, is_server_depot, context|
+  it "uses #{context} depot" do
+    if is_zone_depot
+      @zone.log_file_depot = zone_depot
+    end
+
+    if is_server_depot
+      @miq_server.log_file_depot = server_depot
+    end
+
+    stub_vmdb_util_methods_for_collection_log
+    allow_any_instance_of(LogFile).to receive(:upload)
+
+    @miq_server.post_logs(:taskid => miq_task.id, :context => context)
+
+    log_file_depot = LogFile.first.file_depot
+
+    if context == "Zone"
+      expect(log_file_depot).to eq(zone_depot)
+    else
+      expect(log_file_depot).to eq(server_depot)
+    end
+  end
+end
+
+shared_examples "post_logs_fails" do |is_zone_depot, is_server_depot, context|
+  it "raises error 'Log depot settings not configured'" do
+    if is_zone_depot
+      @zone.log_file_depot = zone_depot
+    end
+
+    if is_server_depot
+      @miq_server.log_file_depot = server_depot
+    end
+
+    stub_vmdb_util_methods_for_collection_log
+    allow_any_instance_of(LogFile).to receive(:upload)
+
+    expect do
+      @miq_server.post_logs(:taskid => miq_task.id, :context => context)
+    end.to raise_error(RuntimeError, "Log depot settings not configured")
+  end
+end
+
 describe MiqServer do
   context "LogManagement" do
+    let(:server_depot) { FactoryGirl.create(:file_depot) }
+    let(:zone_depot) { FactoryGirl.create(:file_depot) }
+    let(:miq_task) { FactoryGirl.create(:miq_task) }
+
     before do
       _, @miq_server, @zone = EvmSpecHelper.create_guid_miq_server_zone
       @miq_server2          = FactoryGirl.create(:miq_server, :zone => @zone)
+
+      ENV["APPLIANCE_PG_DATA"] = "/pgdata/path"
```

Comment:
Why's this needed?  Can we stub a method that calls this?  I'd rather not modify ENV because we need to undo it in the after...